napari/components/layerlist.py

```diff
@@ -126,8 +126,30 @@ def _update_name(self, event):
         layer = event.source
         layer.name = self._coerce_name(layer.name, layer)
 
+    def _ensure_unique(self, values, allow=()):
+        bad = set(self._list) - set(allow)
+        values = values if isinstance(values, Iterable) else (values,)
+        for v in values:
```

Comment:
this still consumes the iterable! :) 
so https://github.com/napari/napari/pull/4544#discussion_r878887181 still stands

Reply:
I added a `tuple(values)` earlier, I swear, I don't know what happened to it <,<